pkg/kubelet/kubelet_node_status.go

```diff
@@ -554,3 +571,52 @@ func validateNodeIP(nodeIP net.IP) error {
 	}
 	return fmt.Errorf("Node IP: %q not found in the host's network interfaces", nodeIP.String())
 }
+
+// nodeStatusHasChanged compares the original node and current node's status and
+// returns true if any change happens. The heartbeat timestamp is ignored.
+func nodeStatusHasChanged(originalStatus *v1.NodeStatus, status *v1.NodeStatus) bool {
+	if originalStatus == nil || status == nil {
+		return !apiequality.Semantic.DeepEqual(originalStatus, status)
+	}
+
+	if nodeConditionsHaveChanged(originalStatus.Conditions, status.Conditions) {
+		return true
+	}
+
+	originalConditions := originalStatus.Conditions
```

Comment:
nit: This whole block of code could use some comment, e.g., "Setting the conditions to nil to compare the rest of the node status. Restoring the conditions in the defer block before returns".

Reply:
Done

Reply:
This is done every update frequency, i.e., once every 10 sec with default configuration. I updated it to use the copy now.